single_tech_samples/fabric/feature_engineering_on_fabric/README.md

```diff
@@ -0,0 +1,435 @@
+
+   Go to Fabric workspace homepage and create a new __Environment__ by clicking '+ New' button and selecting `Environment (Preview)` under 'Data Engineering' or 'Data Science' experience.
+
+   ![new](./images/featurestore_5.png)
+   ![env](./images/featurestore_4.png)
+
+   In this environment, you will setup the python environment under 'Public Libraries'. This can be done by adding `azureml-featurestore` package using PyPI, or by clicking 'Add from yml' and selecting `./src/environment/Publiclibrary.yml` from this repo.
+
+   ![pip](./images/featurestore_3.png)
+
+   You will also need to edit the Spark properties. The yaml template with the required properties is available at `./src/environment/sparkProperties.yaml`. Replace the value with the resources that you created.
+
+   ```yaml
+   runtime_version: '1.1'
+   spark_conf:
+     - spark.fsd.client_id: <sp-client-id>
+     - spark.fsd.tenant_id: <sp-tenant-id>
+     - spark.fsd.subscription_id: <subscription-id>
+     - spark.fsd.rg_name: <feature-store-resouce-group>
+     - spark.fsd.name: <feature-store-name>
+     - spark.fsd.fabric.tenant: <fabric-tenant-name> # Fetch from Fabric base URL, like https://<fabric-tenant-name>.powerbi.com/
+     - spark.fsd.purview.account: <purview-account-name>
+   ```
+
+   After this is done, click 'Save' to save the environment, and then 'Publish' to publish it. This may take a few minutes to finish.
+
+   Finally, To apply the environment, you can set the newly created environment as 'default' in the Fabric workspace settings page.
+
+   ![pip](./images/fabric_env_2.png)
+
+   Or you can apply to each specific notebook in the notebook edition page.
+
+   ![pip](./images/fabric_env_1.png)
+
+## Data pipeline setup
+
+This is the main data pipeline which contains all the activities required for data landing, ingestion, cleansing, transformation, and feature registration. Start by creating a new data pipeline in the workspace.
+
+![data_pipeline_01](./images/data_pipeline/data_pipeline_01.png)
+
+### Data landing activity
+
+As mentioned in the introduction, a 'ForEach' activity is used for data landing. This activity downloads multiple files from a public blob storage. Add the 'ForEach' activity to the data pipeline [^1].
+
+![data_pipeline_02](./images/data_pipeline/data_pipeline_02.png)
+
+The configuration of this 'ForEach' activity follows a series of steps as below.
+
+- Add copy data activity
+
+  Select the 'ForEach' activity that was just created and click '+' to add a 'Copy data' activity.
+
+  ![data_pipeline_03](./images/data_pipeline/data_pipeline_03.png)
+
+- Define source and destination for 'Copy data' activity
+
+  Select the 'Copy data' activity and switch to the 'Source' tab. Choose 'external' as the 'Data store type'. For 'Connection', click '+' to create a new HTTP link. On the creation page, enter <https://stmdwpublic.blob.core.windows.net/> as the server URL. Click the 'Create' button to complete the setup.
```

Comment:
Where is this currently hosted? https://stmdwpublic.blob.core.windows.net/

Reply:
It's hosted in our external DTD subscription :-) I have put a lock on the resource group.